### PR TITLE
Week2/testing actions

### PR DESCRIPTION
**Pull request by JakeFantin**
Tests for some actions:
- create
- new
- edit
- update
- show

Some Syntax Errors solved in the process.

### CODE REVIEW

File: spec/requests/articles_spec.rb [hunk 1]
```diff
@@ -5,14 +5,73 @@
...
+      follow_redirect!
+
+      expect(response).to render_template(:show)
+    end
+
+    it "redirects to edit on failure" do
+      article = Article.new(title: "Title", body: "This is a body.")
+      article.save();
+      put "/articles/#{article.id}", :params => { :article => { :title => "Ti" }}
+      expect(response).to render_template(:edit)
```
Comment (JakeFantin) [inline spec/requests/articles_spec.rb:72]: Not sure what to do here in terms of looking for the error that is returned here.

File: spec/requests/articles_spec.rb [hunk 2]
```diff
@@ -3,16 +3,88 @@
 RSpec.describe "Articles", type: :request do
   describe "GET /" do
     it "returns http success" do
-      get "/articles"
+      get "/"
```
Comment (J-pilon) [inline spec/requests/articles_spec.rb]: You could use get root_path to test your route helpers. Just a suggestion.

File: spec/requests/articles_spec.rb (hunk 1 above)
Comment (J-pilon) [reply spec/requests/articles_spec.rb:72]: How about ```expect(response).to have_http_status(422)```

File: spec/requests/articles_spec.rb [hunk 3]
```diff
@@ -3,16 +3,88 @@
...
+  describe "DELETE /destroy" do
+    it "redirect to root on success" do
+      article = Article.new(title: "Title", body: "This is a body.")
+      article.save();
+      delete "/articles/#{article.id}"
+      # expect{Article.find(id: article.id)}.to raise_error(instance_of("ActiveRecord::RecordNotFound"))
+      expect(response).to have_http_status(:redirect)
+      follow_redirect!
+
+      expect(response).to render_template(:index)
```
Comment (J-pilon) [inline spec/requests/articles_spec.rb:84]: Also could check that the database count is reduced by one. ```expect { delete article_path(article) }.to change(Article.count).by(-1).
Then you would check the response and template afterwards.

File: spec/requests/articles_spec.rb (hunk 1 above)
Comment (JakeFantin) [reply spec/requests/articles_spec.rb:72]: oooh nice. thank you.

File: spec/requests/articles_spec.rb (hunk 2 above)
Comment (JakeFantin) [reply spec/requests/articles_spec.rb]: this was really helpful!

File: spec/requests/articles_spec.rb (hunk 3 above)
Comment (JakeFantin) [reply spec/requests/articles_spec.rb:84]: whoa. This was crazy. I had to put Article.count inside of a block (maybe a rails 7 thing?) but worked as desired. Do you know why some 'expect' use cases use blocks and not '()'?

**Pull request merged**